### DISCUSSION TITLE
pwsh 7.3.3 on RHEL 8 core dumps sometimes

### DISCUSSION THREAD

**Discussion by tonyguadagno** [PowerShell/PowerShell#19335] [Q&A]
hi, i have powershell 7.3.3 installed on rhel 8 (current patching).  running pwsh from command line always seems to work, but i am using Nagios and the NCPA agent.  in this case, i have the NCPA agent calling a wrapper application like this:

```
#!/usr/bin/env -S /usr/bin/pwsh -NoProfile
$fqdn = [System.Net.Dns]::GetHostEntry([string]$env:computername).HostName
/usr/local/bin/ps_PHPFPMStatusCheck.ps1 -statusurl http://$fqdn/php-fpm-status
write-host $LASTEXITCODE
exit $LASTEXITCODE
```
running this wrapper from command line works fine also, but when the ncpa agent calls it, i get this:
```
Mar 15 15:37:22 server powershell[765022]: (7.3.3:1:80) [Perftrack_ConsoleStartupStart:PowershellConsoleStartup.WinStart.Informational] PowerShell console is starting up
Mar 15 15:37:22 server powershell[765022]: (7.3.3:B:80) [NamedPipeIPC_ServerListenerStarted:NamedPipe.Open.Informational] PowerShell has started an IPC listening thread on process: 765022 in AppDomain: None.
Mar 15 15:37:22 server powershell[765022]: (7.3.3:1:80) [Perftrack_ConsoleStartupStop:PowershellConsoleStartup.WinStop.Informational] PowerShell console is ready for user input
Mar 15 15:37:22 server kernel: traps: pwsh[765033] general protection fault ip:7f8db5d34929 sp:7f4c8b4d54c8 error:0 in libcoreclr.so[7f8db5c32000+4e0000]
Mar 15 15:37:22 server systemd[1]: Started Process Core Dump (PID 765042/UID 0).
Mar 15 15:37:22 server systemd-coredump[765043]: Resource limits disable core dumping for process 765022 (pwsh).
Mar 15 15:37:22 server systemd-coredump[765043]: Process 765022 (pwsh) of user 977 dumped core.
Mar 15 15:37:22 server systemd[1]: systemd-coredump@32-765042-0.service: Succeeded.
```

i have tried to run from command line logged in as the nagios user that ncpa runs as....that works also.

anyone know why this is happening?  or where to look from here?

thanks in advance.

**Comment by kasini3000** [comment]
The problem is that it happens every time? Or sometimes?

try：
1  #!/ usr/bin/env -S /usr/bin/pwsh -NoProfile
start-sleep -second 1

2  Increasing memory may alleviate the symptoms.

**Comment by tonyguadagno** [comment]
ksasini3000, thanks for replying.  in my environment, i have 4 web servers.  i installed this script on all 4.  Initially, it worked on 3 and failed on 1.  since then, it started failing on a second.  Now, it consistently works on 2 and consistently fails on 2.

i tried the sleep, did not help

when you say "increase memory" do you mean RAM?  i think the box has plenty:
```
free -h
              total        used        free      shared  buff/cache   available
Mem:           15Gi       2.4Gi       7.7Gi       162Mi       5.3Gi        12Gi
Swap:         2.1Gi       4.0Mi       2.1Gi

```

**Comment by tonyguadagno** [comment]
one more piece of data, i increased the sleep to 10 sec...just to make sure this was not the issue.  it still failed..but it failed right away, NOT after 10 sec.  so i guess this is what is causing the core dump?

`#!/usr/bin/env -S /usr/bin/pwsh -NoProfile`

i tried this also, same issue:

`#!/usr/bin/pwsh`

**Comment by tonyguadagno** [comment]
not sure if this helps any, but i changed my wrapper script to be a bash script.  like this

```
#!/usr/bin/bash
/usr/bin/pwsh -file /usr/local/bin/ps_PHPFPMStatusCheck.ps1 -statusuri /php-fpm-status
exit $?
```

running this from command line works find.  but when ncpa agent is told to call it via api call, this happens:

```
Mar 16 10:34:08 server powershell[806740]: (7.3.3:1:80) [Perftrack_ConsoleStartupStart:PowershellConsoleStartup.WinStart.Informational] PowerShell console is starting up
Mar 16 10:34:08 server powershell[806740]: (7.3.3:B:80) [NamedPipeIPC_ServerListenerStarted:NamedPipe.Open.Informational] PowerShell has started an IPC listening thread on process: 806740 in AppDomain: None.
Mar 16 10:34:08 server powershell[806740]: (7.3.3:1:80) [Perftrack_ConsoleStartupStop:PowershellConsoleStartup.WinStop.Informational] PowerShell console is ready for user input
Mar 16 10:34:08 server kernel: traps: pwsh[806750] general protection fault ip:7f038d1d1929 sp:7ec262d824c8 error:0 in libcoreclr.so[7f038d0cf000+4e0000]
Mar 16 10:34:08 server systemd[1]: Started Process Core Dump (PID 806759/UID 0).
Mar 16 10:34:09 server systemd-coredump[806760]: Resource limits disable core dumping for process 806740 (pwsh).
Mar 16 10:34:09 server systemd-coredump[806760]: Process 806740 (pwsh) of user 977 dumped core.
Mar 16 10:34:09 server systemd[1]: systemd-coredump@38-806759-0.service: Succeeded.
```

**Comment by kasini3000** [comment]
I have no idea, perhaps : dnf update kernel, reboot, remove /opt/microsoft/powershell, then reinstall powershell. If it is not a critical server, you can also try shutting down selinux

**Comment by rhubarb-geek-nz** [comment]
> hi, i have powershell 7.3.3 installed on rhel 8 (current patching)

Which build of powershell 7.3.3 are you using, one from the powershell-7.3.3-1.rh.x86_64.rpm , a tar, from packages.microsoft.com or from a different RPM repository? You don't mention if it is arm64 or x64.

**Comment by tonyguadagno** [comment]
hi, I am using the repo from microsoft:

`powershell.x86_64                                  7.3.3-1.rh                                  @packages-microsoft-com-prod`

**Comment by rhubarb-geek-nz** [comment]
Hi, some thoughts, 

RHEL should be flagship Linux distribution for dotnet core, especially when using packages directly from packages.microsoft.com.

The trap is occurring in libcoreclr.so and that is a base component of dotnet itself.

I have had intermittent problems on some platforms which went away by going back to a powershell 7.2.* release. 7.3.3 is using dotnet 7 and the powershell 7.2.* uses dotnet 6 which LTS and a bit longer in the tooth.

I suggest reverting to powershell 7.2.10 and see if the behaviour changes.

In any case, I suggest raising a formal issue here.

**Comment by tonyguadagno** [comment]
hi, i did open this over the weekend
https://github.com/PowerShell/PowerShell/issues/19351

btw, how do i backrev?

**Reply by rhubarb-geek-nz** [reply]
> btw, how do i backrev?

Do a 

```
sudo yum remove powershell
```

And then

```
sudo yum install https://packages.microsoft.com/rhel/8/prod/Packages/p/powershell-7.2.10-1.rh.x86_64.rpm
```

Should do it.

If you have microsoft.packages.com as a registered repository you might want to disable that for a bit to stop a "yum update" from upgrading it back to 7.3.3 again
